scripts/shai-hulud-machine-check.sh

```diff
@@ -0,0 +1,489 @@
+  log_verbose "  ${GREEN}No malicious processes detected${NC}"
+fi
+
+# 3. Check for malware payload files in home directory
+log_verbose ""
+log_verbose "Checking for malware files in home directory..."
+MALWARE_FILES=("setup_bun.js" "bun_environment.js")
+MALWARE_FOUND=false
+for file in "${MALWARE_FILES[@]}"; do
+  if [ -f "$HOME/$file" ]; then
+    log_critical "Malware file: ~/$file"
+    log_verbose "  ${RED}[CRITICAL]${NC} MALWARE FILE FOUND: ~/$file"
+    MALWARE_FOUND=true
+  fi
+done
+
+# Check in common locations
+for dir in "$HOME" "$HOME/Desktop" "$HOME/Downloads" "/tmp"; do
+  if [ -d "$dir" ]; then
+    for file in "${MALWARE_FILES[@]}"; do
+      FOUND=$(find "$dir" -maxdepth 2 -name "$file" -type f 2>/dev/null | head -5)
+      if [ -n "$FOUND" ]; then
+        while IFS= read -r f; do
+          log_critical "Malware file: $f"
+          log_verbose "  ${RED}[CRITICAL]${NC} MALWARE FILE FOUND: $f"
+          MALWARE_FOUND=true
+        done <<< "$FOUND"
+      fi
+    done
+  fi
+done
+
+if [ "$MALWARE_FOUND" = false ]; then
+  log_verbose "  ${GREEN}No malware files detected${NC}"
+fi
+
+# 4. Check for backdoor workflow files
+log_verbose ""
+log_verbose "Checking for backdoor workflow files..."
+BACKDOOR_FOUND=false
+# Search for discussion.yaml in .github/workflows directories under ~/Code or common project locations
+# The malicious workflow contains: runs-on: self-hosted AND ${{ github.event.discussion.body }}
+for base_dir in "$HOME/Code" "$HOME/Projects" "$HOME/src" "$HOME/dev" "$HOME/workspace"; do
+  if [ -d "$base_dir" ]; then
+    FOUND=$(find "$base_dir" -maxdepth 5 -path "*/.github/workflows/discussion.yaml" -type f 2>/dev/null | head -10)
+    if [ -n "$FOUND" ]; then
+      while IFS= read -r f; do
+        # Check if file contains the malicious pattern
+        if grep -q "self-hosted" "$f" 2>/dev/null && grep -q "github.event.discussion" "$f" 2>/dev/null; then
+          log_critical "Backdoor workflow: $f"
+          log_verbose "  ${RED}[CRITICAL]${NC} BACKDOOR WORKFLOW FOUND: $f"
+          log_verbose "  This workflow uses self-hosted runner with discussion body injection."
+          BACKDOOR_FOUND=true
+        else
+          log_high "Unverified discussion.yaml: $f"
+          log_verbose "  ${YELLOW}[HIGH]${NC} UNVERIFIED WORKFLOW: $f"
+          log_verbose "  Found discussion.yaml - please verify this is legitimate."
+        fi
+      done <<< "$FOUND"
+    fi
+  fi
+done
+
+if [ "$BACKDOOR_FOUND" = false ]; then
+  log_verbose "  ${GREEN}No backdoor workflows detected${NC}"
+fi
+
+###########################################
+# HIGH-RISK CHECKS
+###########################################
+
+log_verbose ""
+log_verbose "${BOLD}=== High-Risk Checks ===${NC}"
+log_verbose ""
+
+# 4. Check for exfiltration artifacts
+log_verbose "Checking for exfiltration artifacts..."
+EXFIL_FILES=("cloud.json" "truffleSecrets.json" "environment.json" "actionsSecrets.json")
```

Comment:
Per the [Wiz report](https://www.wiz.io/blog/shai-hulud-2-0-ongoing-supply-chain-attack), `contents.json` is created locally on compromised machines as part of the exfiltration process to collect stolen credentials before uploading to attacker-controlled repositories.